tests/ha/control_pod/test_control_pod_soft_failures.py

```diff
@@ -0,0 +1,617 @@
+
+        LOGGER.info("Step 5: Create IAM user and perform IOs after soft-failure.")
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.test_prefix = 'test-45498-1'
+        self.s3_clean.update(users)
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True,
+                                                    nclients=2, nsamples=2, setup_s3bench=False)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 5: Created IAM user and performed IOs after soft-failure.")
+
+        LOGGER.info("ENDED: Verify IAM user creation/deletion and IOs on already created IAM user "
+                    "while performing control pod soft-failure in loop 1 by 1 for N control pods.")
+
+    # pylint: disable-msg=too-many-statements
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-45499")
+    def test_iam_cruds_soft_failure_all_pods(self):
+        """
+        Test IOs and IAM user CRUDs before and after csm-agent is down in all control pods.
+        (Negative Scenario)
+        """
+        LOGGER.info("STARTED: Test IOs and IAM user CRUDs before and after csm-agent is down in"
+                    " all control pods")
+
+        num_users = HA_CFG["s3_operation_data"]["no_csm_users"]
+
+        if self.scale_req:
+            LOGGER.info("Scale replicas for control pod to %s", self.repl_num)
+            resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
+                                                                deploy=self.deploy_name)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Total number of control pods in the cluster are: %s", self.repl_num)
+
+        LOGGER.info("Adding %s delay in config files of all control pods",
+                    self.max_proc_restart_delay)
+        self.add_proc_restart_delay(self.max_proc_restart_delay)
+        LOGGER.info("Successfully added %s delay in config files of all control pods",
+                    self.max_proc_restart_delay)
+        self.change_proc_delay = True
+
+        LOGGER.info("Step 1: Perform WRITE-READ-Verify on user %s", self.uids[0])
+        self.test_prefix = 'test-45499'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(self.users.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs-READs-Verify with variable sizes objects.")
+
+        LOGGER.info("Create %s iam users for deletion", num_users)
+        del_users = self.mgnt_ops.create_account_users(nusers=num_users)
+
+        LOGGER.info("Step 2: Create soft failures and perform IAM user creation")
+        for pod in self.ctrl_pod_list:
```

Comment:
I assume deleteing container in pod won't chnage the pod name so this looks good.

Reply:
No, it won't change the pod name.